torch/_inductor/fx_passes/fuse_attention.py

```diff
@@ -146,6 +146,32 @@ def _sfdp_replacement_6(query, key, value, attn_mask, dropout_p):
 # TODO(jansel): make these pattern work with lowmem_dropout=True
 
 
+def _sfdp_params_check(match):
+    assert all(k in match.kwargs for k in ("query", "key", "value"))
+    query = match.kwargs["query"].meta["val"]
+    key = match.kwargs["key"].meta["val"]
+    value = match.kwargs["value"].meta["val"]
+    if not (query.dtype == key.dtype == value.dtype) or not (
+        query.device == key.device == value.device
+    ):
+        return False
+    add_mask_node = filter_nodes(match.nodes, aten.add.Tensor)
+    # Has attn_mask add.
+    if len(add_mask_node) > 0:
+        attn_mask_node = add_mask_node[0].args[1]
+        # attn_mask_node may be a float/int number.
+        if not hasattr(attn_mask_node, "meta"):
+            return False
+        attn_mask = attn_mask_node.meta["val"]
+        if (
+            not isinstance(attn_mask, torch.Tensor)
+            or attn_mask.dtype != query.dtype
+            or query.device != attn_mask.device
```

Comment:
Currently, we need to make sure ```attn_mask.dtype == query.dtype```, if we meet ```attn_mask.dtype != query.dtype``` in a real model, we can add dtype conversion before calling ```scaled_dot_product_attention```.